### PR TITLE
Release Zebra (`v2.0.0-rc.0`)

### LINKED ISSUE #8679: Publish Zebra NU6 Mainnet release candidate: (v2.0.0-rc.0)
- This is a release candidate with the mainnet activation heights in the ECC dependencies but not in zebrad.
- End support support should be up to before the NU6 activation height.

# Prepare for the Release

These release steps can be done a week before the release, in separate PRs.
They can be skipped for urgent releases.

## State Full Sync Test

To check consensus correctness, we want to test that the state format is valid after a full sync. (Format upgrades are tested in CI on each PR.)

- [ ] Make sure there has been [at least one successful full sync test](https://github.com/ZcashFoundation/zebra/actions/workflows/ci-integration-tests-gcp.yml?query=event%3Aschedule) since the last state change, or
- [ ] Start a manual workflow run with a Zebra and `lightwalletd` full sync.

State format changes can be made in `zebra-state` or `zebra-chain`. The state format can be changed by data that is sent to the state, data created within the state using `zebra-chain`, or serialization formats in `zebra-state` or `zebra-chain`. 

After the test has been started, or if it has finished already:
- [ ] Ask for a state code freeze in Slack. The freeze lasts until the release has been published.

## Checkpoints

For performance and security, we want to update the Zebra checkpoints in every release.
- [ ] You can copy the latest checkpoints from CI by following [the zebra-checkpoints README](https://github.com/ZcashFoundation/zebra/blob/main/zebra-utils/README.md#zebra-checkpoints).

## Missed Dependency Updates

Sometimes `dependabot` misses some dependency updates, or we accidentally turned them off.

This step can be skipped if there is a large pending dependency upgrade. (For example, shared ECC crates.)

Here's how we make sure we got everything:
- [ ] Run `cargo update` on the latest `main` branch, and keep the output
- [ ] If needed, [add duplicate dependency exceptions to deny.toml](https://github.com/ZcashFoundation/zebra/blob/main/book/src/dev/continuous-integration.md#fixing-duplicate-dependencies-in-check-denytoml-bans)
- [ ] If needed, remove resolved duplicate dependencies from `deny.toml`
- [ ] Open a separate PR with the changes
- [ ] Add the output of `cargo update` to that PR as a comment

# Prepare and Publish the Release

Follow the steps in the [release checklist](https://github.com/ZcashFoundation/zebra/blob/main/.github/PULL_REQUEST_TEMPLATE/release-checklist.md) to prepare the release:

Release PR:
- [ ] Update Changelog
- [ ] Update README
- [ ] Update Zebra Versions
- [ ] Update End of Support Height

Publish Release:
- [ ] Create & Test GitHub Pre-Release
- [ ] Publish GitHub Release
- [ ] Publish Rust Crates
- [ ] Publish Docker Images

## Related work

#3948

### PR DESCRIPTION
**Pull request by oxarbitrage**
---
name: 'Release Checklist Template'
about: 'Checklist to create and publish a Zebra release'
title: 'Release Zebra (version)'
labels: 'A-release, C-trivial, P-Critical :ambulance:'
assignees: ''

---

# Prepare for the Release

- [x] Make sure there has been [at least one successful full sync test](https://github.com/ZcashFoundation/zebra/actions/workflows/ci-integration-tests-gcp.yml?query=event%3Aschedule) since the last state change, or start a manual full sync.
- [x] Make sure the PRs with the new checkpoint hashes and missed dependencies are already merged.
      (See the release ticket checklist for details)

# Summarise Release Changes

These steps can be done a few days before the release, in the same PR:

## Change Log

**Important**: Any merge into `main` deletes any edits to the draft changelog.
Once you are ready to tag a release, copy the draft changelog into `CHANGELOG.md`.

We use [the Release Drafter workflow](https://github.com/marketplace/actions/release-drafter) to automatically create a [draft changelog](https://github.com/ZcashFoundation/zebra/releases). We follow the [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) format.

To create the final change log:
- [x] Copy the **latest** draft changelog into `CHANGELOG.md` (there can be multiple draft releases)
- [x] Delete any trivial changes
    - [x] Put the list of deleted changelog entries in a PR comment to make reviewing easier
- [x] Combine duplicate changes
- [x] Edit change descriptions so they will make sense to Zebra users
- [x] Check the category for each change
  - Prefer the "Fix" category if you're not sure

## README

README updates can be skipped for urgent releases.

Update the README to:
- [x] Remove any "Known Issues" that have been fixed since the last release.
- [x] Update the "Build and Run Instructions" with any new dependencies.
      Check for changes in the `Dockerfile` since the last tag: `git diff <previous-release-tag> docker/Dockerfile`.
- [x] If Zebra has started using newer Rust language features or standard library APIs, update the known working Rust version in the README, book, and `Cargo.toml`s

You can use a command like:
```sh
fastmod --fixed-strings '1.58' '1.65'
```

## Create the Release PR

- [x] Push the updated changelog and README into a new branch
      for example: `bump-v1.0.0` - this needs to be different to the tag name
- [x] Create a release PR by adding `&template=release-checklist.md` to the comparing url ([Example](https://github.com/ZcashFoundation/zebra/compare/bump-v1.0.0?expand=1&template=release-checklist.md)).
- [x] Freeze the [`batched` queue](https://dashboard.mergify.com/github/ZcashFoundation/repo/zebra/queues) using Mergify.
- [x] Mark all the release PRs as `Critical` priority, so they go in the `urgent` Mergify queue.
- [x] Mark all non-release PRs with `do-not-merge`, because Mergify checks approved PRs against every commit, even when a queue is frozen.

# Update Versions and End of Support

### Choose a Release Level

Zebra follows [semantic versioning](https://semver.org). Semantic versions look like: MAJOR.MINOR.PATCH[-TAG.PRE-RELEASE]

Choose a release level for `zebrad`. Release levels are based on user-visible changes from the changelog:
- Mainnet Network Upgrades are `major` releases
- significant new features or behaviour changes; changes to RPCs, command-line, or configs; and deprecations or removals are `minor` releases
- otherwise, it is a `patch` release

Zebra's Rust API doesn't have any support or stability guarantees, so we keep all the `zebra-*` and `tower-*` crates on a beta `pre-release` version.

### Update Crate Versions

If you're publishing crates for the first time, [log in to crates.io](https://github.com/ZcashFoundation/zebra/blob/doc-crate-own/book/src/dev/crate-owners.md#logging-in-to-cratesio),
and make sure you're a member of owners group.

Check that the release will work:
- [x] Update crate versions, commit the changes to the release branch, and do a release dry-run:

```sh
# Update everything except for alpha crates and zebrad:
cargo release version --verbose --execute --allow-branch '*' --workspace --exclude zebrad --exclude zebra-scan --exclude zebra-grpc beta
# Due to a bug in cargo-release, we need to pass exact versions for alpha crates:
cargo release version --verbose --execute --allow-branch '*' --package zebra-scan 0.1.0-alpha.4
cargo release version --verbose --execute --allow-branch '*' --package zebra-grpc 0.1.0-alpha.2
# Update zebrad:
cargo release version --verbose --execute --allow-branch '*' --package zebrad patch # [ major | minor | patch ]
# Continue with the release process:
cargo release replace --verbose --execute --allow-branch '*' --package zebrad
cargo release commit --verbose --execute --allow-branch '*'
```

Crate publishing is [automatically checked in CI](https://github.com/ZcashFoundation/zebra/actions/workflows/release-crates-io.yml) using "dry run" mode, however due to a bug in `cargo-release` we need to pass exact versions to the alpha crates:

- [x] Update `zebra-scan` and `zebra-grpc` alpha crates in the [release-crates-dry-run workflow script](https://github.com/ZcashFoundation/zebra/blob/main/.github/workflows/scripts/release-crates-dry-run.sh)
- [x] Push the above version changes to the release branch.

## Update End of Support

The end of support height is calculated from the current blockchain height:
- [x] Find where the Zcash blockchain tip is now by using a [Zcash explorer](https://zcashblockexplorer.com/blocks) or other tool.
- [x] Replace `ESTIMATED_RELEASE_HEIGHT` in [`end_of_support.rs`](https://github.com/ZcashFoundation/zebra/blob/main/zebrad/src/components/sync/end_of_support.rs) with the height you estimate the release will be tagged.

<details>

<summary>Optional: calculate the release tagging height</summary>

- Add `1152` blocks for each day until the release
- For example, if the release is in 3 days, add `1152 * 3` to the current Mainnet block height

</details>

## Update the Release PR

- [x] Push the version increments and the release constants to the release branch.

# Publish the Zebra Release

## Create the GitHub Pre-Release

- [x] Wait for all the release PRs to be merged
- [x] Create a new release using the draft release as a base, by clicking the Edit icon in the [draft release](https://github.com/ZcashFoundation/zebra/releases)
- [x] Set the tag name to the version tag,
      for example: `v1.0.0`
- [x] Set the release to target the `main` branch
- [x] Set the release title to `Zebra ` followed by the version tag,
      for example: `Zebra 1.0.0`
- [x] Replace the prepopulated draft changelog in the release description with the final changelog you created;
      starting just _after_ the title `## [Zebra ...` of the current version being released,
      and ending just _before_ the title of the previous release.
- [x] Mark the release as 'pre-release', until it has been built and tested
- [x] Publish the pre-release to GitHub using "Publish Release"
- [x] Delete all the [draft releases from the list of releases](https://github.com/ZcashFoundation/zebra/releases)

## Test the Pre-Release

- [x] Wait until the Docker binaries have been built on `main`, and the quick tests have passed:
    - [x] [ci-unit-tests-docker.yml](https://github.com/ZcashFoundation/zebra/actions/workflows/ci-unit-tests-docker.yml?query=branch%3Amain)
    - [x] [ci-integration-tests-gcp.yml](https://github.com/ZcashFoundation/zebra/actions/workflows/ci-integration-tests-gcp.yml?query=branch%3Amain)
- [ ] Wait until the [pre-release deployment machines have successfully launched](https://github.com/ZcashFoundation/zebra/actions/workflows/cd-deploy-nodes-gcp.yml?query=event%3Arelease)

## Publish Release

- [x] [Publish the release to GitHub](https://github.com/ZcashFoundation/zebra/releases) by disabling 'pre-release', then clicking "Set as the latest release"

## Publish Crates

- [ ] [Run `cargo login`](https://github.com/ZcashFoundation/zebra/blob/doc-crate-own/book/src/dev/crate-owners.md#logging-in-to-cratesio)
- [ ] Run `cargo clean` in the zebra repo (optional)
- [ ] Publish the crates to crates.io: `cargo release publish --verbose --workspace --execute`
- [ ] Check that Zebra can be installed from `crates.io`:
      `cargo install --locked --force --version 1.minor.patch zebrad && ~/.cargo/bin/zebrad`
      and put the output in a comment on the PR.

## Publish Docker Images
- [ ] Wait for the [the Docker images to be published successfully](https://github.com/ZcashFoundation/zebra/actions/workflows/release-binaries.yml?query=event%3Arelease).
- [ ] Un-freeze the [`batched` queue](https://dashboard.mergify.com/github/ZcashFoundation/repo/zebra/queues) using Mergify.
- [ ] Remove `do-not-merge` from the PRs you added it to

## Release Failures

If building or running fails after tagging:

<details>

<summary>Tag a new release, following these instructions...</summary>

1. Fix the bug that caused the failure
2. Start a new `patch` release
3. Skip the **Release Preparation**, and start at the **Release Changes** step
4. Update `CHANGELOG.md` with details about the fix
5. Follow the release checklist for the new Zebra version

</details>

### CODE REVIEW

**Comment by oxarbitrage** [comment]
The following pull requests were not included in the changelog:


### Trivial *TODO:* put this in a PR comment, not the CHANGELOG
- release(checkpoints): Update checkpoints for `2.0.0.-rc.0` ([#8924](https://github.com/ZcashFoundation/zebra/pull/8924))
- fix(clippy): last lints ([#8855](https://github.com/ZcashFoundation/zebra/pull/8855))
- Update utilities README.md ([#8824](https://github.com/ZcashFoundation/zebra/pull/8824))
- add(docs): Add minimal hardware requirements ([#8822](https://github.com/ZcashFoundation/zebra/pull/8822))
- fix(deps): Replace `serde_yaml` by `serde_yml` ([#8825](https://github.com/ZcashFoundation/zebra/pull/8825))
- build(deps): bump docker/scout-action from 1.13.0 to 1.14.0 in the devops group ([#8890](https://github.com/ZcashFoundation/zebra/pull/8890))
- build(deps): bump the devops group with 3 updates ([#8919](https://github.com/ZcashFoundation/zebra/pull/8919))
- build(deps): bump the devops group with 4 updates ([#8905](https://github.com/ZcashFoundation/zebra/pull/8905))
- build(deps): bump the prod group across 1 directory with 24 updates ([#8899](https://github.com/ZcashFoundation/zebra/pull/8899))
- build(deps): bump tj-actions/changed-files from 45.0.0 to 45.0.1 in the devops group ([#8835](https://github.com/ZcashFoundation/zebra/pull/8835))
- build(deps): bump tj-actions/changed-files from 45.0.1 to 45.0.2 in the devops group ([#8874](https://github.com/ZcashFoundation/zebra/pull/8874))

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,43 @@ All notable changes to Zebra are documented in this file.
...
+
+Please note that support for this release candidate is expected to conclude prior to the NU6 activation heights.
+
+### Security
+
+- Added Docker Scout vulnerabilities scanning ([#8871](https://github.com/ZcashFoundation/zebra/pull/8871))
+
+### Added
+
+- For regtest mode only, added `generate` ([#8849](https://github.com/ZcashFoundation/zebra/pull/8849)) and `stop` ([#8839](https://github.com/ZcashFoundation/zebra/pull/8839), [#8863](https://github.com/ZcashFoundation/zebra/pull/8863)) RPC methods
```
Comment (arya2) [inline CHANGELOG.md]:
```suggestion
- Added Regtest-only `generate` and `stop` RPC methods ([#8849](https://github.com/ZcashFoundation/zebra/pull/8849), [#8839](https://github.com/ZcashFoundation/zebra/pull/8839), [#8863](https://github.com/ZcashFoundation/zebra/pull/8863))
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,43 @@ All notable changes to Zebra are documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org).
 
+## [Zebra 2.0.0-rc.0](https://github.com/ZcashFoundation/zebra/releases/tag/v2.0.0-rc.0) - 2024-10-11
+
+This version is a release candidate for the Zcash NU6 network upgrade on the Mainnet. While this version does not yet include the NU6 Mainnet activation heights, all required functionality and tests are in place.
```
Comment (arya2) [inline CHANGELOG.md]:
```suggestion
This version is a release candidate for the Zcash NU6 network upgrade on the Mainnet. While this version does not yet include the NU6 Mainnet activation height or current protocol version, all required functionality and tests are in place.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,43 @@ All notable changes to Zebra are documented in this file.
...
+Please note that support for this release candidate is expected to conclude prior to the NU6 activation heights.
+
+### Security
+
+- Added Docker Scout vulnerabilities scanning ([#8871](https://github.com/ZcashFoundation/zebra/pull/8871))
+
+### Added
+
+- For regtest mode only, added `generate` ([#8849](https://github.com/ZcashFoundation/zebra/pull/8849)) and `stop` ([#8839](https://github.com/ZcashFoundation/zebra/pull/8839), [#8863](https://github.com/ZcashFoundation/zebra/pull/8863)) RPC methods
+- Added more fields to `getmininginfo` RPC method ([#8860](https://github.com/ZcashFoundation/zebra/pull/8860))
```
Comment (arya2) [inline CHANGELOG.md]:
```suggestion
- Added fields to `getmininginfo` RPC method response ([#8860](https://github.com/ZcashFoundation/zebra/pull/8860))
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -5,6 +5,43 @@ All notable changes to Zebra are documented in this file.
...
+- Moved the python RPC tests framework from zcashd into Zebra ([#8866](https://github.com/ZcashFoundation/zebra/pull/8866))
```
Comment (arya2) [inline CHANGELOG.md]:
```suggestion
- Copied the Python RPC test framework from zcashd into Zebra ([#8866](https://github.com/ZcashFoundation/zebra/pull/8866))
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -5,6 +5,43 @@ All notable changes to Zebra are documented in this file.
...
+
+### Changed
+
+- Regtest halving interval to match zcashd and added a configurable halving interval for custom testnets ([#8888](https://github.com/ZcashFoundation/zebra/pull/8888), [#8928](https://github.com/ZcashFoundation/zebra/pull/8928))
+- Updates post-NU6 Major Grants funding stream address on Mainnet ([#8914](https://github.com/ZcashFoundation/zebra/pull/8914))
+- Upgrade ECC dependencies to match zcashd `v6.0.0-rc1` ([#8918](https://github.com/ZcashFoundation/zebra/pull/8918))
```
Comment (arya2) [inline CHANGELOG.md]: Subjective: This one might not be user-visible.

```suggestion
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -5,6 +5,43 @@ All notable changes to Zebra are documented in this file.
...
+
+### Fixed
+
+- Multiple Docker fixes ([#8870](https://github.com/ZcashFoundation/zebra/pull/8870), [#8827](https://github.com/ZcashFoundation/zebra/pull/8827), [#8817](https://github.com/ZcashFoundation/zebra/pull/8817), [#8796](https://github.com/ZcashFoundation/zebra/pull/8796))
```
Comment (arya2) [inline CHANGELOG.md]: Only the first one seems user-visible

```suggestion
- Remove debugging output by default in Docker image ([#8870](https://github.com/ZcashFoundation/zebra/pull/8870))
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -5,6 +5,43 @@ All notable changes to Zebra are documented in this file.
...
+- Actions fixes ([#8910](https://github.com/ZcashFoundation/zebra/pull/8910), [#8893](https://github.com/ZcashFoundation/zebra/pull/8893), [#8868](https://github.com/ZcashFoundation/zebra/pull/8868), [#8865](https://github.com/ZcashFoundation/zebra/pull/8865))
```
Comment (arya2) [inline CHANGELOG.md]:
```suggestion
- Fixes a typo in configuration file path of the docker-compose file ([#8893](https://github.com/ZcashFoundation/zebra/pull/8893))
```

File: zebrad/src/components/sync/end_of_support.rs [hunk 8]
```diff
@@ -13,7 +13,7 @@ use zebra_chain::{
 use crate::application::release_version;
 
 /// The estimated height that this release will be published.
-pub const ESTIMATED_RELEASE_HEIGHT: u32 = 2_626_500;
+pub const ESTIMATED_RELEASE_HEIGHT: u32 = 2_678_363;
```
Comment (oxarbitrage) [inline zebrad/src/components/sync/end_of_support.rs:16]: When i changed this the tip was `2_677_211`, i am giving 1 day more (+`1152`) = `2_678_363`

File: zebrad/src/components/sync/end_of_support.rs [hunk 9]
```diff
@@ -22,8 +22,8 @@ pub const ESTIMATED_RELEASE_HEIGHT: u32 = 2_626_500;
 ///
 /// - Zebra will exit with a panic if the current tip height is bigger than the `ESTIMATED_RELEASE_HEIGHT`
 ///   plus this number of days.
-/// - Currently set to 14 weeks.
-pub const EOS_PANIC_AFTER: u32 = 70;
+/// - Currently set to 5 weeks to end support before Mainnet Nu6 activation at block [`2_726_400`](https://zips.z.cash/zip-0253).
+pub const EOS_PANIC_AFTER: u32 = 35;
```
Comment (oxarbitrage) [inline zebrad/src/components/sync/end_of_support.rs:26]: Nu6 activation height is in about 41 days from now:

2726400 - 2678363 = 48037 % 1152 = 41,6

I changed the support of this release to 5 weeks to end about 1 week before the activation height. I am open to other alternatives.

**Review by oxarbitrage — COMMENTED** [review]
Comments about the end of support changes for this release.

**Review by upbqdn — APPROVED** [review]

**Pull request merged**